lib/basedriver/helpers.js

```diff
@@ -202,6 +208,66 @@ async function testZipArchive (zipPath) {
   }
 }
 
+async function copyFromWindowsNetworkShare (app, appExt, mountroot, windowsUserName, windowsPassword) {
+  if (system.isWindows()) {
+    return await copyLocallyFromWindowsShare(app, appExt);
+  } else {
+    return await mountWindowsShareOnMac(app, mountroot, windowsUserName, windowsPassword);
+  }
+}
+
+async function mountWindowsShareOnMac (app, mountroot, windowsUserName, windowsPassword) {
+  let pathSplit = app.split("\\");
+  let networkShare = pathSplit[2];
+  let rootFolder = pathSplit[3];
+  app = app.replace(/\\/g, "/");
+  app = app.replace(`/${networkShare}`, mountroot);
+  let mountPath = `/${mountroot}/${rootFolder}`;
+  let mountNetworkShare = async function () {
+    await fs.mkdir(mountPath);
+    let mountArgs = [`-t`, `smbfs`, `${windowsUserName}:${windowsPassword}@${networkShare}/${rootFolder}`, `${mountPath}`];
+    try {
+      await exec('mount', mountArgs, { maxBuffer: 524288 });
+    } catch (err) {
+      logger.errorAndThrow(`Error mounting: ${err.message}`);
+    }
+  };
+  if (await fs.exists(mountPath)) {
+    if (await fs.exists(app)) {
+      return app;
+    } else {
+      let umountArgs = [`${mountPath}`];
+      try {
+        await exec('umount', umountArgs, { maxBuffer: 524288 });
+      } catch (err) {
+        logger.error(`Error Unmounting :${err.message}`);
+      }
+      await fs.rimraf(mountroot);
+      await mountNetworkShare();
+    }
+  } else {
+    await mountNetworkShare();
+  }
+  return app;
+}
+
+async function copyLocallyFromWindowsShare (app, appExt) {
+  let fileInfo = await tempDir.open({prefix: 'appium-app', suffix: appExt});
+  return await copyFile(app, fileInfo.path);	
+}
+
+async function copyFile (source, target) {
```

Comment:
Does `fs.copyFile` from [appium-support](https://github.com/appium/appium-support/blob/master/lib/fs.js#L34) not work?